pandas/core/strings.py

```diff
@@ -478,7 +478,10 @@ def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=None):
     flags : int, default 0 (no flags)
         re module flags, e.g. re.IGNORECASE
     na : default NaN, fill value for missing values.
-    as_indexer : DEPRECATED
+    as_indexer : False, by default, gives deprecated behavior better
+        achieved using str_extract. True return boolean indexer.
```

Comment:
This is not fully correct. It is just ignored, so the explanation for True or False doesn't matter.